en/debug/jtag_pins.md

```diff
@@ -0,0 +1,130 @@
+
+> **Note** The **Pixhawk Debug Port** also exposes a UART for access to the [System Console](../debug/system_console.md) (a console that provides low-level access to the system, debug output and analysis of the system boot process).
+
+<!--
+### FMUv2
+
+Does this have a debug port? where is spec
+Is it a Hirose DF13
+
+
+### FMUv3
+
+Does this have a debug port? where is spec
+Is it a Hirose DF13
+-->
+
+### FMUv4/FMUv5
+
+FMU V4,5 has a 6 pin JST GH
+
+[Pixracer](http://docs.px4.io/master/en/flight_controller/pixracer.html#pinouts): [Pixhawk debug port](https://pixhawk.org/pixhawk-connector-standard/#dronecode_debug) (JST SM06B connector)
+
+Pin | Signal | Volt
+--- | --- | ---
+1 (red) | VCC TARGET SHIFT | +3.3V
+2 (blk) | CONSOLE TX (OUT) | +3.3V
+3 (blk) | CONSOLE RX (IN) | +3.3V
+4 (blk) | SWDIO | +3.3V
+5 (blk) | SWCLK | +3.3V
+6 (blk) | GND | GND
+
+
+### FMUv5X
+
+FMUv5X will have a [10-pin JST SUR](https://order.jst-mfg.com/InternetShop/app/index.php?back=2&product=53574000&jgcd=0#showProductDetail).
+It adds JTAG pins: SWO, nRST. TRACECLK, TRACED0 Pins 7 and 8 are GPIOS to gate timing or 1 wire trace (full trace is on).
+<!-- is this link/example correct type for connector? -->
+
+FMU Debug Port Pinout
+
+Pin | Signal | Volt
+--- | --- | ---
+1 (red) | FMU VDD | +3.3V
+2 (blk) | CONSOLE TX (OUT) | +3.3V
+3 (blk) | CONSOLE RX (IN) | +3.3V
+4 (blk) | SWDIO | +3.3V
+5 (blk) | SWCLK | +3.3V
+6 (blk) | SWO | +3.3V
+7 (blk) | TRACECLK | +3.3V
+8 (blk) | TRACED0 | +3.3V
+9 (blk) | nRST | +3.3V
+10 (blk) | GND | GND
+
+
+> **Note** This board will have an additional 8-pin JST SUR for ARM ETM TRACE).
+
+
+## Hex Cube
+
+Cube debug port information can be found here: [Debug ports](http://docs.px4.io/master/en/flight_controller/pixhawk-2.html#debug-ports).
+
+If pulled apart the Cube has a 6 pin JST SUR connector (marked as J10) for both FMU and IO debugging.
+
+<!--
+Does the numbering match all the other 6 pin port examples.
+What label is the IO port marked as?
+So obviously it is much easier to just connect to the carrier board port. Who would know what interface that is. PhilipR?
+-->
+
+
+## Kakute F7
+
+There are Test points on the board.
+<!-- I take that to mean that there are SWDIO etc pins somewhere. How do I find out where? -->
+
+UART3 RX and TX are configured for System console. 
+
+
+## Hovergames RDDRONE-FMUK66
+
+The Hovergames RDDRONE-FMUK66 has a similar debug interface to Pixhawk (it supports an additional JTAG pin - nRST) and uses a JST-GH connector). 
+The debug interface is documented here: [RDDRONE-FMUK66 Debug interface](https://nxp.gitbook.io/hovergames/rddrone-fmuk66/connectors/debug-interface-dcd-lz)
+
+![Debug RDDRONE-FMUK66 using Jlink and debug interface](../../assets/debug/jlink_hovergames.png)
+<!-- what am I seeing here? Ie the jlink and the FC are obvious. 
+But what is the orange connector between them, and what is the connector out of the center of them doing? -->
+
+
+<!-- 
```

Comment:
@davids5 This is somewhat an aside. When do you use each debug port? ie is it that you expect people to know exactly what is loaded into FMU vs IO and process is same, for each, you just connect to a different port?

Reply:
IO - load new bootloader or IO Firmware and debug IO
FMU load new bootloader or FMU Firmware and debug OR Just for the Console.